### PR TITLE
Added possibility to add Constructor-Expectations on hard dependancies

### PR DESCRIPTION
**Pull request by patzerr**
Hi,
at some point when writing code one has to use the "new"-keyword (in my case on the https://packagist.org/packages/predis/predis Module) to retrieve a class.

With this patch one is able to add Expectations on what constructor parameters will be supplied to those classes.

hth

### CODE REVIEW

**Comment by davedevelopment** [comment]
I don't think I'll ever use it, but I'm generally  :+1:

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,7 +25,7 @@
 * BC BREAK - Fix Mockery not trying default expectations if there is any concrete expectation
 * BC BREAK - Mockery's PHPUnit integration will mark a test as risky if it
   thinks one it's exceptions has been swallowed in PHPUnit > 5.7.6. Use `$e->dismiss()` to dismiss.
-
+* Added possibility to add Constructor-Expections on hard dependancies, read: Mockery::mock('overload:...')
```
Comment (Bilge) [inline CHANGELOG.md]: *dependancies* isn't a word. Applies throughout this PR.

Comment (patzerr) [reply CHANGELOG.md]: Please provide a suitable replacement.

At least my search bubble yielded some interesting results:
https://en.oxforddictionaries.com/definition/dependency
> ‘the islands became dependencies of Norway and Denmark’
https://askubuntu.com/questions/361741/what-are-dependencies
> By dependencies are missing the package manager...

hth

Comment (davedevelopment) [reply CHANGELOG.md]: @Bilge means it is spelt incorrectly, you have the right word. It has 3 Es, 0 As.

Comment (patzerr) [reply CHANGELOG.md]: lol now i see it, thanks @davedevelopment

**Comment by patzerr** [comment]
hi again,

> I don't think I'll ever use it, but I'm generally

The Prophecy Mocking Framework allows this for example by doing lazy mock doubles (that require a call to the "reveal"-Method)

e.g.
`$client->willBeConstructedWith([$config['url'], $config]);`

**Comment by robertbasic** [comment]
@patzerr hey, thanks for the PR, would you mind fixing the conflicts and bringing your PR up to date with the current master? Thanks!

**Pull request merged**